diff --git a/my_patch.patch b/my_patch.patch
new file mode 100644
index 0000000..e69de29
diff --git a/src/user/user.controller.js b/src/user/user.controller.js
index f13f93b..e1bfa8f 100644
--- a/src/user/user.controller.js
+++ b/src/user/user.controller.js
@@ -3,8 +3,13 @@ const userService = require("./user.service");
 exports.getLogin = (req, res) => {
   res.render("login.html");
 };
-exports.postLogin = (req, res) => {
-  const result = userService.postLogin();
-  console.log(result);
-  res.redirect("/");
+exports.postLogin = async (req, res, next) => {
+  try {
+    const result = await userService.postLogin();
+    console.log(result);
+    res.redirect("/");
+  } catch (error) {
+    // throw new Error("Controller Error " + error.message);
+    next();
+  }
 };
diff --git a/src/user/user.repository.js b/src/user/user.repository.js
index a66d92a..ba5850d 100644
--- a/src/user/user.repository.js
+++ b/src/user/user.repository.js
@@ -1,3 +1,15 @@
-exports.findOneByUserInfo = (user_id, user_pw) => {
-  return true;
+const pool = require("../../pool");
+
+exports.findOneByUserInfo = async (user_email, user_password) => {
+  // const sql = `SELECT * FROM users WHERE email="${user_email}" and password=${user_password}`;
+  // 위처럼 쓰지 않고 sql injection 등의 보안 취약점을 방지하기 위하여 prepared statement 방식으로 써야한다.
+  try {
+    // --> async 함수는 이렇게 try, catch 구문으로 감싸서 에러처리를 반드시 해줘라
+    const sql = `SELECT * FROM users WHERE email=? and password=?`;
+    const [[result]] = await pool.query(sql, [user_email, user_password]);
+    // console.log(result);
+    return result;
+  } catch (error) {
+    throw new Error("Repository Error " + error.message);
+  }
 };
diff --git a/src/user/user.service.js b/src/user/user.service.js
index f6ed4e4..2ded7c1 100644
--- a/src/user/user.service.js
+++ b/src/user/user.service.js
@@ -1,6 +1,13 @@
 const userRepository = require("./user.repository");
 
-exports.postLogin = () => {
-  const result = userRepository.findOneByUserInfo("web7722", "1234");
-  return result;
+exports.postLogin = async () => {
+  try {
+    const result = await userRepository.findOneByUserInfo(
+      "web7722@gmail.com",
+      "1234"
+    );
+    return result;
+  } catch (error) {
+    throw new Error("Service Error " + error.message);
+  }
 };
